### PR TITLE
fix: nix fix for wasm-unknown-unknown target for wasm-pack

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
- Fixes error using wasm-pack from nix compared to what rustup provides. 
- Also run cargo-test in ci for rust-version of workspace directly via manifest-path
- proper test release for wasm side.

### CODE REVIEW

File: rust+wasm/{{project-name}}-wasm/README.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ Javascript-wrapper file, the {{project-name}}-wasm README (and version), and a
 For running tests in the current directly, use this command:
 
 ```console
-wasm-pack test
+wasm-pack test --node | --headless --*
```
Comment (bgins) [inline rust+wasm/{{project-name}}-wasm/README.md:97]: Having trouble running this command. Should there be a `|` in it?

When I remove the pipe, it produces this error:

> Error: The `--headless` flag only applies to browser tests. Node does not provide a UI, so it doesn't make sense to talk about a headless version of Node tests.

Comment (zeeshanlakhani) [reply rust+wasm/{{project-name}}-wasm/README.md:97]: it's not a real command haha.

Comment (zeeshanlakhani) [reply rust+wasm/{{project-name}}-wasm/README.md:97]: i'll fix up the readme to make the explicit ones haha.

File: rust+wasm/{{project-name}}-wasm/README.md [hunk 2]
```diff
@@ -94,7 +94,7 @@ Javascript-wrapper file, the {{project-name}}-wasm README (and version), and a
 For running tests in the current directly, use this command:
```
Comment (bgins) [inline rust+wasm/{{project-name}}-wasm/README.md:94]: Not actually in the current PR, but noticed a possible typo here. Should this be "current directory"?

**Review by bgins — APPROVED** [review]
Looks good! Nice to have that enforcement for the conventional commits. 📏 🎉

**Pull request merged**